src/typing_extensions.py

```diff
@@ -1958,3 +1959,86 @@ def decorator(cls_or_fn):
 if not hasattr(typing, "TypeVarTuple"):
     typing._collect_type_vars = _collect_type_vars
     typing._check_generic = _check_generic
+
+
+if sys.version_info >= (3, 11):
+    NamedTuple = typing.NamedTuple
+else:
+    def _caller(depth=1, default='__main__'):
```

Comment:
`_caller()` is only ever called without arguments in our code.

Reply:
Good point!